client/src/components/ProtectedRoute/ProtectedRoute.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { Redirect, Route, RouteComponentProps } from 'react-router-dom';
+import { ProtectedRouteProps } from '../../interface/Route';
+
+/**
+ * Protected route component for handling routing
+ * based on user authentication status.
+ * @param ProtectedRouteProps
+ * @returns {JSX.Element}
+ */
+const ProtectedRoute = ({ component: Component, token, ...rest }: ProtectedRouteProps): JSX.Element => {
+  return (
+    <Route
+      {...rest}
```

Comment:
I do see some duplicated code here, I think you forgot to base the branch properly, but it shouldn't be a problem since your protected route PR just got approved! 😄

Reply:
Your right I didn't base it from the protected route branch. I will merge the updates into this branch.